src/Mapping/Request/BasicEntity.php

```diff
@@ -78,7 +80,13 @@ protected function normalize(string $property, $value)
 	 */
 	protected function fromBodyRequest(ApiRequest $request): self
 	{
-		return $this->factory((array) $request->getJsonBody(true));
+		try {
+			$body = (array) $request->getJsonBody(true);
+		} catch (JsonException $ex) {
+			throw new ClientErrorException('Invalid json data: ' . $ex->getMessage());
```

Comment:
What about `throw new ClientErrorException('Invalid json data', null, $e);`? To keep origin message.

Reply:
Thank you for your comment, it is incorporated